### PR TITLE
Improvements to internationalisation docs

### LINKED ISSUE #612: Document new approach to multi lingual sites
Currently, the documentation contains a tutorial about how to create multi lingual sites by duplicating your site tree.

Wagtail 0.6+ lets you use a different approach using Djangos ``i18n_patterns`` and ``LocaleMiddleware`` which is better in some ways as you can duplicate the fields instead of duplicating all pages. We should document this approach as well.

### PR DESCRIPTION
**Pull request by kaedroho**
Fixes #612 
Refs: #96

This pull request adds documentation for a better method of creating a multi lingual site with Wagtail

### CODE REVIEW

**Comment by kaedroho** [comment]
Thanks @agelinas 

> My only concern about your well written document on some of the techniques one can use to make Wagtail CMS multilingual is that it seems to me like Wagtail isn't going to include some of those functions in its code base, like Django-CMS did, but I could be wrong. 

I'm in favour of adding new functions into Wagtail to make multilingual sites easier (and I think the other core devs would be in favour too).

It can take a while to add new APIs to Wagtail as they need to be well thought through, agreed upon, well tested/documented, etc. So I just wanted to improve the docs with what Wagtail has now, and hopefully we can cut it down later as we add new i18n features.

File: docs/advanced_topics/i18n/index.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+ - Make sure `USE_I18N <https://docs.djangoproject.com/en/1.8/ref/settings/#use-i18n>`_ is set to ``True``
+ - Set `LANGUAGE_CODE <https://docs.djangoproject.com/en/1.8/ref/settings/#std:setting-LANGUAGE_CODE>`_ to your websites' primary language
+
+If there is a translation available for your language, the Wagtail admin backend should now be in the language you've chosen.
+
+
+Creating sites with multiple languages
+======================================
+
+You can create sites with multiple language support by levaraging Django's `translation features <https://docs.djangoproject.com/en/1.8/topics/i18n/translation/>`_.
```
Comment (tomdyson) [inline docs/advanced_topics/i18n/index.rst]: levaraging -> leveraging

File: docs/advanced_topics/i18n/index.rst [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+-----------------------------------------------
+
+Just enabling the multi-language support in Django sometimes may not be enough. By default, Django will serve different languages of the same page with the same URL. This has a couple of drawbacks:
+
+ - Users cannot change language without changing their browser settings
+ - It may not work well with various caching setups (as content varies based on browser settings)
+
+Django's ``i18n_patterns`` feature, when enabled, prefixes the URLs with the language code (eg ``/en/about-us``). Users are forwarded to their preferred version, based on browser language, when they first visit the site.
+
+This feature is enabled through the projects root URL configuration. Just put the views you would like to have this enabled for in an ``i18n_patterns`` list and append that to the other url patterns:
```
Comment (tomdyson) [inline docs/advanced_topics/i18n/index.rst]: projects -> project's

Comment (tomdyson) [inline docs/advanced_topics/i18n/index.rst]: url -> URL

File: docs/advanced_topics/i18n/index.rst [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    urlpatterns += i18n_patterns('',
+        # These URLs will have /<language_code>/ appended to the beginning
+
+        url(r'^search/$', 'search.views.search', name='search'),
+
+        url(r'', include(wagtail_urls)),
+    )
+
+You can implement switching between languages by changing the part at the beginning of the URL. As each language has it's own URL, it also works well with just about any caching setup.
```
Comment (tomdyson) [inline docs/advanced_topics/i18n/index.rst]: it's > its

**Pull request merged**